wp-cli/civicrm.php

```diff
@@ -25,1253 +25,1305 @@
+			}
+
+			if ( ! $dbpass = $this->getOption( 'dbpass', false ) ) {
+				return WP_CLI::error( 'CiviCRM database password not specified.' );
+			}
+
+			if ( ! $dbhost = $this->getOption( 'dbhost', false ) ) {
+				return WP_CLI::error( 'CiviCRM database host not specified.' );
+			}
+
+			if ( ! $dbname = $this->getOption( 'dbname', false ) ) {
+				return WP_CLI::error( 'CiviCRM database name not specified.' );
+			}
+
+			if ( $lang = $this->getOption( 'lang', false ) and ! $langtarfile = $this->getOption( 'langtarfile', false ) ) {
+				return WP_CLI::error( 'CiviCRM language tarfile not specified.' );
+			}
+
+			# begin install
+
+			if ( $plugin_path = $this->getOption( 'destination', false ) ) {
+				$plugin_path = ABSPATH . $plugin_path;
+			} else {
+				$plugin_path = WP_PLUGIN_DIR . '/civicrm';
+			}
+
+			global $crmPath;
+			$crmPath = "$plugin_path/civicrm";
+			$crm_files_present = is_dir( $crmPath );
+
+			# extract the archive
+			if ( $this->getOption( 'tarfile', false ) ) {
+				# should probably never get to here as Wordpress Civi comes in a zip file, but
+				# just in case that ever changes ..
+				if ( $crm_files_present ) {
+					return WP_CLI::error( 'Existing CiviCRM found. No action taken.' );
+				}
+
+				if ( ! $this->untar( dirname( $plugin_path ) ) ) {
+					return WP_CLI::error( 'Error extracting tarfile' );
+				}
+			} elseif ( $this->getOption( 'zipfile', false ) ) {
+				if ( $crm_files_present ) {
+					return WP_CLI::error( 'Existing CiviCRM found. No action taken.' );
+				}
+
+				if ( ! $this->unzip( dirname( $plugin_path ) ) ) {
+					return WP_CLI::error( 'Error extracting zipfile' );
+				}
+			} elseif ( $crm_files_present ) {
+				// Site is already extracted (which is how we're running this
+				// script); we just need to run the installer.
+
+			} else {
+				return WP_CLI::error( 'No zipfile specified, use --zipfile=path/to/zipfile or extract file ahead of time' );
+			}
+
+			# include civicrm installer helper file
+			$civicrm_installer_helper = "$crmPath/install/civicrm.php";
+
+			if ( ! file_exists( $civicrm_installer_helper ) ) {
+				return WP_CLI::error( 'Archive could not be unpacked OR CiviCRM installer helper file is missing.' );
+			}
+
+			if ( $crm_files_present ) {
+				// We were using a directory that was already there.
+				WP_CLI::success( 'Using installer files found on the site.' );
+			} else {
+				// We must've just unpacked the archive because it wasn't there
+				// before.
+				WP_CLI::success( 'Archive unpacked.' );
+			}
+			require_once $civicrm_installer_helper;
+
+			if ( '' != $lang ) {
+				if ( ! $this->untar( $plugin_path, 'langtarfile' ) ) {
+					return WP_CLI::error( 'No language tarfile specified, use --langtarfile=path/to/tarfile' );
+				}
+			}
+
+			# create files dirs
+			$upload_dir      = wp_upload_dir();
+			$settings_dir     = $upload_dir['basedir'] . DIRECTORY_SEPARATOR . 'civicrm' . DIRECTORY_SEPARATOR;
+			civicrm_setup( "$settings_dir" );
```

Comment:
This creates a directory structure of wp-content/uploads/civicrm/civicrm/

I would change to:
```
$userFilesDir     = $upload_dir['basedir'] . DIRECTORY_SEPARATOR;
civicrm_setup( "$userFilesDir" );
```

We also get a wp-content/uploads/civicrm/files/ directory structure that does not happen in a normal install.   While that will work, we should track that down to make the install consistent.